src/runtime/contrib/vitis_ai/vitis_ai_runtime.cc

```diff
@@ -0,0 +1,147 @@
+  model_path_ = model_path;
+  target_ = target;
+  xgraph_ = load_xgraph_model(model_path_);
+  in_tensor_names_ = xgraph_->get_input_names();
+  out_tensor_names_ =  xgraph_->get_meta_attr("tvm_out_tensors").get_strings();
+  pyxir::partition(xgraph_, std::vector<std::string>{target}, "");
+  pyxir::RunOptionsHolder run_options(new pyxir::runtime::RunOptions());
+  run_options->on_the_fly_quantization = true;
+  rt_mod_ = pyxir::build_rt(xgraph_, target_ , in_tensor_names_, out_tensor_names_,
+                                                                "vai", run_options);
+}
+
+
+Module VitisAIRuntimeCreate(const std::string& name,
+                            const std::string& model_path,
+                            const std::string& target) {
+  Array<String> const_vars;
+  auto exec = make_object<VitisAIRuntime>(name, const_vars);
+  exec->Init(model_path, target);
+  return Module(exec);
+}
+
+
+
+TVM_REGISTER_GLOBAL("tvm.vitis_ai_runtime.create").set_body([](TVMArgs args, TVMRetValue* rv) {
+  *rv = VitisAIRuntimeCreate(args[0], args[1], args[2]);
+});
+
+Module VitisAIRuntimeLoadFromBinary(void* strm ) {
+    dmlc::Stream* stream = static_cast<dmlc::Stream*>(strm);
+
+    std::string model_path;
+    std::string symbol_name;
+    std::vector<std::string> const_vars;
+    std::string target;
+    stream->Read(&model_path);
+    stream->Read(&target);
+    stream->Read(&symbol_name);
+    stream->Read(&const_vars);
+    Array<String> const_names;
+    for (const auto& it : const_vars) {
+      const_names.push_back(it);
+    }
+    auto exec = make_object<VitisAIRuntime>(symbol_name, const_names);
+    exec->Init(model_path, target);
+    return Module(exec);
+  }
+
+TVM_REGISTER_GLOBAL("runtime.module.loadbinary_VitisAIRuntime")
+                  .set_body_typed(VitisAIRuntimeLoadFromBinary);
+
+void VitisAIRuntime::SaveToBinary(dmlc::Stream* stream)  {
+  stream->Write(this-> model_path_);
+  stream->Write(this-> target_);
+  stream->Write(this->symbol_name_);
+  std::vector<std::string> consts;
+  for (const auto& it : const_names_) {
+    consts.push_back(it);
+    }
+  stream->Write(consts);
+  }
+
+
+PackedFunc VitisAIRuntime::GetFunction(const std::string& name,
+                                      const ObjectPtr<Object>& sptr_to_self) {
+  if (name == "get_symbol") {
+      return PackedFunc(
+          [sptr_to_self, this](TVMArgs args, TVMRetValue* rv) { *rv = this->symbol_name_; });
+    } else if (name == "get_const_vars") {
+      return PackedFunc(
+          [sptr_to_self, this](TVMArgs args, TVMRetValue* rv) { *rv = this->const_names_; });
+    } else if ("__init_" + this->symbol_name_ == name) {
+      // The function to initialize constant tensors.
+      return PackedFunc([sptr_to_self, this](TVMArgs args, TVMRetValue* rv) {
+        CHECK_EQ(args.size(), 1U);
+        this->initialized_ = true;
+        *rv = 0;
+      });
+    } else {
```

Comment:
what's the handle for this? It seems this is for execution.  I think you need to have a check here, e.g. `if (name == "xxx")`. Otherwise, we may return a wrong packed function.